### DISCUSSION TITLE
Trusted Clusters not working

### DISCUSSION THREAD

**Discussion by sanchezzjf** [gravitational/teleport#33012] [Q&A] [answered]
Hello, I'm trying to connect a trusted cluster for my teleport community, one of them(the root) is responsible for accessing the machines and the other one(leaf) for accessing applications, when I try to join the root cluster I get this INFO and the connection doesn't work:

ALPN connection upgrade test failed for "teleport.ex.com:3024": dial tcp 192.168.0.1:3024: connect: connection refused. client/alpn_conn_upgrade.go:87

There is no rule on the firewall that block the communication between the two teleports and the self-signed certificate installed on the root one is also installed on the leaf one.

What am I doing wrong?

**Comment by webvictim** [comment, marked as answer]
You're probably using multiplexing on your root cluster, which means that all traffic goes over port 443 and port 3024 won't be used for incoming tunnel connections.

You can't modify trusted cluster configurations after saving them, so here's what you should do:
- Add a new trusted cluster join token if your current one has expired - `sudo tctl tokens add --type=trusted_cluster` on your root cluster
- Edit the existing trusted cluster configuration and copy the YAML out of the configuration window
- Delete the existing trusted cluster configuration
- Select the options to create a new trusted cluster
- Paste in the YAML you copied above, but make a couple of changes:
  - Remove the line for `tunnel_addr` completely - just leave `web_proxy_addr` as `teleport.ex.com:443`
  - Change the `token` value to the new trusted cluster join token you generated above
- Save the trusted cluster configuration and it should work.

**Reply by sanchezzjf** [reply]
I did that and no errors were prompted, but when I do "tsh clusters" in both servers nothing appears
Log: 
```shell
out 05 12:42:41 teleport-app teleport[6445]: 2023-10-05T12:42:41Z INFO [AUTH]      Creating enabled Trusted Cluster relationship. trusted_cluster:teleport-sec auth/trustedcluster.go:128
out 05 12:42:41 teleport-app teleport[6445]: 2023-10-05T12:42:41Z INFO [AUTH]      Sending validate request; token=************************d5d9e8e1, CAs=[CA(name=tp.ex.com, type=host)] auth/trustedc>
out 05 12:42:41 teleport-app teleport[6445]: 2023-10-05T12:42:41Z INFO [AUTH]      Received validate response; CAs=[CA(name=teleport-sec, type=host) CA(name=teleport-sec, type=user) CA(name=teleport-sec,>
out 05 12:42:41 teleport-app teleport[6445]: 2023-10-05T12:42:41Z INFO [AUDIT]     trusted_cluster.create cluster_name:tp.ex.com code:T7000I ei:0 event:trusted_cluster.create expires:0001-01-01T00:0>
out 05 12:47:21 teleport-app teleport[6445]: 2023-10-05T12:47:21Z INFO [CA]        Generating TLS certificate 1.3.9999.1.15=#13046e6f6e65,1.3.9999.1.9=#130c31302e3230302e35302e3130,1.3.9999.1.7=#130e7470>
out 05 12:47:21 teleport-app teleport[6445]: 2023-10-05T12:47:21Z INFO [AUDIT]     cert.create cert_type:user cluster_name:tp.ex.com code:TC000I ei:0 event:cert.create client_ip:10.200.50.10 expires>
``` 
When I tsh clusters on the leaf:
```shell
root@teleport-app:~# tsh clusters
Cluster Name   Status Cluster Type Labels Selected
-------------- ------ ------------ ------ --------
tp.ex.com online root                *

root@teleport-app:~#
```
root:
```shell
root@teleport-sec:~# tsh clusters
Cluster Name Status Cluster Type Labels Selected
------------ ------ ------------ ------ --------
teleport-sec online root                *

root@teleport-sec:~#
```

Trusted Clusters File:
```yaml
kind: trusted_cluster
metadata:
  id: 1696509761151663087
  name: teleport-sec
spec:
  enabled: true
  role_map:
  - local:
    - visitor
    remote: access
  token: 8dc5b9b8500471aff345527fd5d9e8e1
  tunnel_addr: teleport.root.com:443
  web_proxy_addr: teleport.root.com:443
version: v2
```

**Reply by webvictim** [reply]
Do you see both clusters in the dropdown at the top left of the web UI? 
![image](https://github.com/gravitational/teleport/assets/897821/a85ef2f6-a8a3-4e84-aebc-cf6a3e949e1a)

Try logging out and logging again if not.

Also `tsh logout`, then `tsh login` again and try `tsh clusters` again.

**Reply by sanchezzjf** [reply]
No it doesn't appear in either dropdowns and tsh clusters remains the same, the only thing that shows is in the leaf cluster trusted clusters
![image](https://github.com/gravitational/teleport/assets/81991756/9ee2eb94-86a7-41ef-99b2-147479625bbd)
the only sign of a connection

**Reply by sanchezzjf** [reply]
It worked, it was not showing for me because of some testing role I created, I asked for a coworker to test and it worked just fine, thank you
